src/Chapter1.hs

```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
After looking at your code, I think that you are not using this extension 🙂 
Did you have any issues, so you started to work with some more advanced stuff? Maybe you have any questions?